nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ListDatabaseTables.java

```diff
@@ -260,8 +241,27 @@ public void onTrigger(ProcessContext context, ProcessSession session) throws Pro
                         .filter(segment -> !StringUtils.isEmpty(segment))
                         .collect(Collectors.joining("."));
 
-                String fqTableName = stateMapProperties.get(fqn);
-                if (fqTableName == null) {
+                String lastTimestampForTable = stateMapProperties.get(fqn);
+                boolean refreshTable = true;
+                try {
+                    // Refresh state if the interval has elapsed
+                    long lastRefreshed = -1;
+                    final long currentTime = System.currentTimeMillis();
+                    if (!StringUtils.isEmpty(lastTimestampForTable)) {
+                        lastRefreshed = Long.parseLong(lastTimestampForTable);
+                    }
+                    if (lastRefreshed == -1 || (refreshInterval > 0 && currentTime >= (lastRefreshed + refreshInterval))) {
+                        stateMapProperties.remove(lastTimestampForTable);
+                    } else {
+                        refreshTable = false;
+                    }
+                } catch (final NumberFormatException nfe) {
+                    getLogger().error("Failed to retrieve observed last table fetches from the State Manager. Will not perform "
```

Comment:
This exception really *shouldn't* happen since we are setting the long ourselves but if it does it will fail entirely until state is cleared by the user. In addition to returning and yielding, it should probably clear the offending state entry (and log in the error message that this is hapenning and the ramifications). This will at least give the processor a chance to continue working if it every reaches this state.